### PR TITLE
Add e2e test for server-url validation

### LINKED ISSUE #5666: Show error if default value for Rancher server-url is a local host IP
**Detailed Description**
Provide insight to the user in case of communication issues with DO instance.

- Add validation to the server URL property both on the setup screen and the global settings page that lets you change it - when set to 127.0.0.1 or localhost, we show a warning banner.
- Text for banner should be: `If the Server URL is internal to the Rancher server (e.g. localhost) the downstream clusters may not be able to communicate with Rancher`

**Context**
While interacting with a cloud service, if the `server-url` is not set, requests will hang unfulfilled without a clear message, especially for beginner users.

Displaying a notice if Rancher has not a remote IP, we should inform the user of possible issue, e.g. creation of Clusters.

Configuration path: `/c/local/settings/management.cattle.io.setting`
Area: Global Settings -> Settings -> server-url

### PR DESCRIPTION
**Pull request by momesgin**
### Summary
Fixes #5666 & #8893 

### Checklist
- [ ] The PR is linked to an issue and the linked issue has a Milestone, or no issue is needed
- [ ] The PR has a Milestone 
- [ ] The PR template has been filled out
- [ ] The PR has been self reviewed 
- [ ] The PR has a reviewer assigned
- [ ] The PR has automated tests or clear instructions for manual tests and the linked issue has appropriate QA labels, or tests are not needed
- [ ] The PR has reviewed with UX and tested in light and dark mode, or there are no UX changes

### CODE REVIEW

File: cypress/e2e/po/edit/settings.po.ts [hunk 1]
```diff
@@ -36,6 +36,22 @@ export default class SettingsEditPo extends PagePo {
...
+  serverUrlLocalhostWarningBanner() {
+    return this.self().find('[data-testid="setting-serverurl-localhost-warning"]');
+  }
+
+  serverUrlTrailingForwardSlashErrorBanner() {
+    return this.self().find('[data-testid="Server URL should not have a trailing forward slash."]');
+  }
+
+  serverUrlNonHttpsErrorBanner() {
+    return this.self().find('[data-testid="Server URL must be https."]');
```
Comment (izaac) [inline cypress/e2e/po/edit/settings.po.ts:48]: @yonasberhe23 do you know if it's possible to add a conditional testid depending on the banner error, or assign a testid per banner? I'd prefer to use a locator rather find by text if possible.

Comment (yonasberhe23) [reply cypress/e2e/po/edit/settings.po.ts:48]: Yes, it looks like the locator is the label text in this case. The locator dynamically changes based on the error label. 

I do think these banner methods can be consolidated into one that accepts the label as an argument. Also, it should be using the `BannersPo`

e.g.
```
  bannerContent(label: string) {
    return new BannersPo('[data-testid="banner-content"]', this.self()).bannerElement(`${ label }`);
  }
```

Comment (momesgin) [reply cypress/e2e/po/edit/settings.po.ts:48]: @yonasberhe23 yes, that's right.

I'll update the code to reflect your suggestions.

**Comment by momesgin** [comment]
@izaac @yonasberhe23 I've updated the tests based on the feedback, I've also added the same validation tests to the `rancher-setup` page.

**Review by izaac — APPROVED** [review]
I think this is now implementing what @yonasberhe23 commented. And using fixtures was a good call too.

**Pull request merged**